pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -562,13 +562,21 @@ func (r *ReconcileClusterDeployment) reconcile(request reconcile.Request, cd *hi
 	if !controllerutils.HasFinalizer(cd, hivev1.FinalizerDeprovision) {
 		cdLog.Debugf("adding clusterdeployment finalizer")
 		if err := r.addClusterDeploymentFinalizer(cd); err != nil {
-			cdLog.WithError(err).Log(controllerutils.LogLevel(err), "error adding finalizer")
+			cdLog.WithError(err).Log(controllerutils.LogLevel(err), "error adding deprovision finalizer")
 			return reconcile.Result{}, err
 		}
 		metricClustersCreated.WithLabelValues(hivemetrics.GetClusterDeploymentType(cd)).Inc()
 		return reconcile.Result{}, nil
 	}
 
+	if cd.Spec.ClusterPoolRef != nil && cd.Spec.ClusterPoolRef.ClusterDeploymentCustomizationRef != nil && !controllerutils.HasFinalizer(cd, hivev1.FinalizerCustomizationRelease) {
+		cdLog.Debugf("adding customization finalizer")
+		if err := r.addClusterDeploymentCustomizationFinalizer(cd); err != nil {
```

Comment:
> nit: These long method names are bugging me out -- this ain't Java :P

Thank you! I hate it too but I tried to follow the style in the project but got stuck with horrible class name.

> Since we're already in the clusterdeployment package, perhaps we could do without ClusterDeployment in the names. So e.g.

Sounds great to me :+1: